File: src/main/java/com/googongill/aditory/controller/dto/user/SignoutResponse.java
package com.googongill.aditory.controller.dto.user;

import com.fasterxml.jackson.annotation.JsonAutoDetect;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import lombok.Builder;

@Builder
@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
public class SignoutResponse {
    private Long userId;
    private String username;

    public static SignoutResponse of(Long userId, String username) {
        return SignoutResponse.builder()
                .userId(userId)
                .username(username)
                .build();
    }
}

File: src/main/java/com/googongill/aditory/controller/dto/user/response/SignoutResponse.kt
package com.googongill.aditory.controller.dto.user.response

import com.fasterxml.jackson.annotation.JsonAutoDetect
import com.fasterxml.jackson.databind.annotation.JsonSerialize

@JsonSerialize
@JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
data class SignoutResponse(
    val userId: Long?,
    val username: String
) {

    companion object {
        fun of(userId: Long?, username: String): SignoutResponse {
            return SignoutResponse(
                userId = userId,
                username = username
            )
        }
    }

}
